features/guidance/notification.feature

```diff
@@ -0,0 +1,65 @@
+@routing  @guidance @mode-change
+Feature: Notification on turn onto mode change
+
+    Background:
+        Given the profile "car"
+        Given a grid size of 20 meters
+
+    Scenario: Turn onto a Ferry
+        Given the node map
+            |   |   |   |   |   | f |
+            | b | c |   |   | d | e |
+            |   |   |   |   |   |   |
+            | a |   |   |   |   | g |
+
+        And the ways
+            | nodes | highway | route | name  |
+            | ab    | primary |       | bingo |
+            | bc    | primary |       | dingo |
+            | cde   |         | ferry | ferry |
+            | eg    | primary |       | wingo |
+
+        When I route I should get
+            | waypoints | route                   | turns                                                   | modes                               |
+            | a,g       | bingo,ferry,wingo,wingo | depart,notification straight,notification right,arrive  | driving,ferry,driving,driving       |
+
+    Scenario: Turn onto a Ferry
+        Given the node map
+            |   |   |   |   |   |   | g |
+            |   | a |   |   |   |   | e |
+            |   |   | c |   |   | d |   |
```

Comment:
ferry lines sharing coordinates are more or less evil to me. If this should actually be a reasonable modelling in OSM, we need to think about to handle them specially. But staying on a ferry should not emit an instruction.